app/src/main/java/com/naman14/timberx/ui/adapters/FolderAdapter.kt

```diff
@@ -17,51 +17,46 @@ package com.naman14.timberx.ui.adapters
 import android.app.Activity
 import android.graphics.drawable.Drawable
 import android.os.AsyncTask
-import android.os.Environment
-import android.preference.PreferenceManager
+import android.os.Environment.DIRECTORY_MUSIC
+import android.os.Environment.getExternalStoragePublicDirectory
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.ImageView
 import android.widget.TextView
-
-import com.naman14.timberx.repository.FoldersRepository
-import com.naman14.timberx.models.Song
-
-import java.io.File
-import java.util.ArrayList
-
-import androidx.core.content.ContextCompat
+import androidx.core.content.ContextCompat.getDrawable
 import androidx.core.content.edit
 import androidx.recyclerview.widget.RecyclerView
 import com.naman14.timberx.R
+import com.naman14.timberx.models.Song
+import com.naman14.timberx.repository.FoldersRepository
 import com.naman14.timberx.repository.SongsRepository
 import com.naman14.timberx.util.Utils
-import com.naman14.timberx.util.toSongIDs
+import com.naman14.timberx.util.extensions.defaultPrefs
+import com.naman14.timberx.util.extensions.toSongIds
 import com.squareup.picasso.Picasso
+import java.io.File
+
+private const val KEY_LAST_FOLDER = "last_folder"
 
 class FolderAdapter(private val mContext: Activity) : RecyclerView.Adapter<FolderAdapter.ItemHolder>() {
 
-    private var mFileSet: List<File>? = null
-    private val mSongs: MutableList<Song>
-    private var mRoot: File? = null
-    private val mIcons: Array<Drawable>
-    private var mBusy = false
+    private val mIcons: Array<Drawable> = arrayOf(
+            getDrawable(mContext, R.drawable.ic_folder_open_black_24dp)!!,
+            getDrawable(mContext, R.drawable.ic_folder_parent_dark)!!,
+            getDrawable(mContext, R.drawable.ic_file_music_dark)!!,
+            getDrawable(mContext, R.drawable.ic_timer_wait)!!
+    )
 
-    private val LAST_FOLDER = "last_folder"
-    private val prefs = PreferenceManager.getDefaultSharedPreferences(mContext)
-    private val root = prefs.getString(LAST_FOLDER, Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_MUSIC).path)
+    private var files: List<File>? = null
```

Comment:
out of curiousity, is there any reason why this is initialized to nullable instead of `Collections.emptyList()` (after which it could always be forced to be non-null)?

Reply:
@Zhuinden I didn't, this is pretty existing code - but out of scope for this PR

Reply:
Also Kotlin has `emptyList()` outside of Java's `Collections`